kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/DocumentationUrls.java

```diff
@@ -0,0 +1,145 @@
+ * found, the string unknown is used.
+ */
+public enum DocumentationUrls {
+
+    /**
+     * Javadoc home (for Inspector and CSS Analyzer properties)
+     */
+    JAVADOC_HOME("https://openjfx.io/javadoc/11/"),
+    
+    /**
+     * This URL is where you go when the user takes Scene Builder Help action (shortcut F1)
+     */
+    ORACLE_DOCUMENTATION("https://docs.oracle.com/javafx/index.html"),
+    
+    /**
+     * Gluon javadoc home (for Inspector and CSS Analyzer properties)
+     */
+    GLUON_JAVADOC_HOME("https://docs.gluonhq.com/charm/javadoc/latest/"),
+    GLUON_SCENEBUILDER_HOME("https://gluonhq.com/products/scene-builder/"),
+
+    OPENJFX_GETTING_STARTED("https://openjfx.io/openjfx-docs/"),
+    OPENJFX_JAVADOC_HOME("https://openjfx.io/javadoc/{javafx.version.major}/"),
+    OPENJFX_CSS_REFERENCE("https://openjfx.io/javadoc/{javafx.version.major}/javafx.graphics/javafx/scene/doc-files/cssref.html"),
+    OPENJFX_FXML_REFERENCE("https://openjfx.io/javadoc/{javafx.version.major}/javafx.fxml/javafx/fxml/doc-files/introduction_to_fxml.html"),
+
+    GLUON_SCENEBUILDER_CONTRIBUTE("https://github.com/gluonhq/scenebuilder");
+
+    private static final String UNKNOWN = "unknown";
+
+    private static String javaFxMajorVersion = null;
+    
+    private final String url;
+    
+    private DocumentationUrls(String defaultValue) {
+        if (defaultValue != null) {
+            assert !defaultValue.isBlank();
+        }
+        url = defaultValue;
+    }
+    
+    /**
+     * Provides the desired URL with place holders resolved to their values.
+     */
+    @Override
+    public String toString() {
+        if (javaFxMajorVersion == null) {
+            javaFxMajorVersion = getMajorJavaFxVersion(getJavaFxVersion());
+        }
+        return resolveJavaFxVersion(url, javaFxMajorVersion);
+    }
+
+    String getConfiguredValue() {
+        return url;
+    }
+
+    private static String resolveJavaFxVersion(String url, String majorFxVersion) {
+        String placeHolder = "{javafx.version.major}";
+        if (url.contains(placeHolder)) {
+            return url.replace(placeHolder, majorFxVersion);
+        }
+        return url;
+    }
+
+    /**
+     * Detects the used JavaFX version from system properties using
+     * {@code System.getProperty("javafx.version")}. In case this property is
+     * undefined, the String "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    static String getJavaFxVersion() {
+        return getJavaFxVersion(System.getProperties());
+    }
+
+    /**
+     * Detects the used JavaFX version from given properties instance using property
+     * key {@code "javafx.version"}. In case this property is undefined, the String
+     * "unknown" is provided.
+     * 
+     * @return JavaFX version number or "unknown" when property is undefined
+     */
+    static String getJavaFxVersion(Properties systemProperties) {
```

Comment:
If these are package-private _just_ because of testing, I would argue that they should be private and we should only test the public API since there is no other code in production that would call these methods.

Continuing with that reasoning, I wonder if we can just do `System.getProperty(...)`. Then we don't have to null check and we possibly don't event need `Properties`.

Reply:
I agree in removing all those checks as it is all not public API.

Reply:
This method no longer exists in this way.